os/drivers/seclink/seclink_drv_common.c

```diff
@@ -55,6 +57,22 @@ int hd_handle_common_request(int cmd, unsigned long arg, void *lower)
 	case SECLINKIOC_DEINIT:
 		req->res = se->ops->deinit();
 		break;
+#ifdef CONFIG_ARM_TRUSTZONE
+	case SECLINKIOC_ALLOC:
+		if (!info) {
+			return -EINVAL;
+		}
+		params = (hal_init_param *)req->params;
+		req->res = se->ops->allocate_scontext(info->s_size);
```

Comment:
Actually driver doesn't know which lower is being added, so if there's unimplemented operation of sec_ops_s, error will be happen.
Hence, all case in this file should follow below style, you have to consider expandability.
```
if (se->ops->allocate_scontext != NULL) {
      req->res = se->ops->allocate_scontext(info->s_size);
} else {
      req->res = -ENOSYS;
}
```

Reply:
agree with you. I did not put check since the check for unimplemented operation of sec_ops_s is not checked in the existing code. I will add them anways